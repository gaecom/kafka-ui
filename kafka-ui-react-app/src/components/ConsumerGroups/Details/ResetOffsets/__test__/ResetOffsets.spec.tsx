import React from 'react';
import { Route, StaticRouter } from 'react-router';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { render } from 'lib/testHelpers';
import { clusterConsumerGroupResetOffsetsPath } from 'lib/paths';
import { consumerGroupPayload } from 'redux/reducers/consumerGroups/__test__/fixtures';
import ResetOffsets from 'components/ConsumerGroups/Details/ResetOffsets/ResetOffsets';
import fetchMock from 'fetch-mock';

const clusterName = 'cluster1';
const { groupId } = consumerGroupPayload;

const selectresetTypeAndPartitions = async (resetType: string) => {
  userEvent.selectOptions(screen.getByLabelText('Reset Type'), resetType);
  userEvent.click(screen.getByText('Select...'));
  await waitFor(() => {
    userEvent.click(screen.getByText('Partition #0'));
  });
};

const renderComponent = () =>
  render(
    <StaticRouter
      location={{
        pathname: clusterConsumerGroupResetOffsetsPath(
          clusterName,
          consumerGroupPayload.groupId
        ),
      }}
    >
      <Route
        path={clusterConsumerGroupResetOffsetsPath(
          ':clusterName',
          ':consumerGroupID'
        )}
      >
        <ResetOffsets />
      </Route>
    </StaticRouter>
  );

describe('ResetOffsets', () => {
  xit('renders progress bar for initial state', () => {
    renderComponent();
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  describe('with consumer group', () => {
    it('calls resetConsumerGroupOffsets', async () => {
      fetchMock.getOnce(
        `/api/clusters/${clusterName}/consumer-groups/${groupId}`,
        consumerGroupPayload
      );
      renderComponent();
      const form = await screen.findByRole('form');
      console.log(form);

      await selectresetTypeAndPartitions('EARLIEST');
      await waitFor(() => {
        userEvent.click(screen.getByText('Submit'));
      });
      // expect(mockResetConsumerGroupOffsets).toHaveBeenCalledTimes(1);
      // expect(mockResetConsumerGroupOffsets).toHaveBeenCalledWith(
      //   'testCluster',
      //   'testGroup',
      //   expectedOutputs.EARLIEST
      // );
    });
  });

  describe('on submit', () => {
    describe('with the default ResetType', () => {});

    // describe('with the ResetType set to LATEST', () => {
    //   it('calls resetConsumerGroupOffsets', async () => {
    //     const mockResetConsumerGroupOffsets = jest.fn();
    //     render(
    //       setupWrapper({
    //         resetConsumerGroupOffsets: mockResetConsumerGroupOffsets,
    //       })
    //     );
    //     await selectresetTypeAndPartitions('LATEST');
    //     await waitFor(() => {
    //       userEvent.click(screen.getByText('Submit'));
    //     });
    //     expect(mockResetConsumerGroupOffsets).toHaveBeenCalledTimes(1);
    //     expect(mockResetConsumerGroupOffsets).toHaveBeenCalledWith(
    //       'testCluster',
    //       'testGroup',
    //       expectedOutputs.LATEST
    //     );
    //   });
    // });

    // describe('with the ResetType set to OFFSET', () => {
    //   it('calls resetConsumerGroupOffsets', async () => {
    //     const mockResetConsumerGroupOffsets = jest.fn();
    //     render(
    //       setupWrapper({
    //         resetConsumerGroupOffsets: mockResetConsumerGroupOffsets,
    //       })
    //     );
    //     await selectresetTypeAndPartitions('OFFSET');
    //     await waitFor(() => {
    //       fireEvent.change(screen.getAllByLabelText('Partition #0')[1], {
    //         target: { value: '10' },
    //       });
    //     });
    //     await waitFor(() => {
    //       userEvent.click(screen.getByText('Submit'));
    //     });
    //     expect(mockResetConsumerGroupOffsets).toHaveBeenCalledTimes(1);
    //     expect(mockResetConsumerGroupOffsets).toHaveBeenCalledWith(
    //       'testCluster',
    //       'testGroup',
    //       expectedOutputs.OFFSET
    //     );
    //   });
    // });

    // describe('with the ResetType set to TIMESTAMP', () => {
    //   it('adds error to the page when the input is left empty', async () => {
    //     const mockResetConsumerGroupOffsets = jest.fn();
    //     render(setupWrapper());
    //     await selectresetTypeAndPartitions('TIMESTAMP');
    //     await waitFor(() => {
    //       userEvent.click(screen.getByText('Submit'));
    //     });
    //     expect(mockResetConsumerGroupOffsets).toHaveBeenCalledTimes(0);
    //     expect(
    //       screen.getByText("This field shouldn't be empty!")
    //     ).toBeInTheDocument();
    //   });
    // });
  });
});
