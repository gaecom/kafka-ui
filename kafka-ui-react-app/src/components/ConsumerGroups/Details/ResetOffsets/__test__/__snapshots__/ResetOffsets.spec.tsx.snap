// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetOffsets on initial render matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="NzFxt"
    >
      <h1>
        Reset offsets
      </h1>
      <div />
    </div>
    <div
      class="hiwAsE"
    >
      <form>
        <div
          class="jOFwJk"
        >
          <div>
            <label
              class="gWUTpy"
              for="topic"
            >
              Topic
            </label>
            <div
              class="select-wrapper jXyqPy"
            >
              <select
                class="kTZcVy"
                id="topic"
                name="topic"
              >
                <option
                  value="__amazon_msk_canary"
                >
                  __amazon_msk_canary
                </option>
                <option
                  value="other_topic"
                >
                  other_topic
                </option>
              </select>
            </div>
          </div>
          <div>
            <label
              class="gWUTpy"
              for="resetType"
            >
              Reset Type
            </label>
            <div
              class="select-wrapper jXyqPy"
            >
              <select
                class="kTZcVy"
                id="resetType"
                name="resetType"
              >
                <option
                  value="EARLIEST"
                >
                  EARLIEST
                </option>
                <option
                  value="LATEST"
                >
                  LATEST
                </option>
                <option
                  value="TIMESTAMP"
                >
                  TIMESTAMP
                </option>
                <option
                  value="OFFSET"
                >
                  OFFSET
                </option>
              </select>
            </div>
          </div>
          <div>
            <label
              class="gWUTpy"
            >
              Partitions
            </label>
            <div
              class="rmsc multi-select"
            >
              <div
                aria-labelledby="Select partitions"
                aria-readonly="true"
                class="dropdown-container"
                tabindex="0"
              >
                <div
                  class="dropdown-heading"
                >
                  <div
                    class="dropdown-heading-value"
                  >
                    <span
                      class="gray"
                    >
                      Select...
                    </span>
                  </div>
                  <svg
                    class="dropdown-heading-dropdown-arrow gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <path
                      d="M6 9L12 15 18 9"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="cmcyoH"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</body>
`;
